cpp/pipelines/add_subsystem_tags.cc

```diff
@@ -425,27 +424,100 @@ void AddSubsystemTags(MARC::Reader * const marc_reader, MARC::Writer * const mar
 }
 
 
+void ExtractAuthors(MARC::Reader * const marc_reader, std::map<std::string, std::set<std::string>> * authors,
+                        const std::unordered_set<std::string> &bible_studies_gnd_numbers,
+                        const std::unordered_set<std::string> &canon_law_gnd_numbers) {
+    static std::vector<std::string> tags_to_check{ "100", "110", "111", "700", "710", "711" };
+    unsigned record_count(0);
+    while (const MARC::Record record = marc_reader->read()) {
+        ++record_count;
+        for (auto tag_to_check : tags_to_check) {
+            for (auto &field : record.getTagRange(tag_to_check)) {
+                const auto subfields(field.getSubfields());
```

Comment:
I wonder why there is no getSubfields(subfield_code) function yet... but it should be a good idea to add one to MARC.h and shorten this code a bit.

Reply:
Maybe including a second parameter to ignore empty subfields to shorten this even more

Reply:
Or you could implement a function like getFirstSubfieldWithCodeAndPrefix